Game/Source/ThirdPersonShooter/BigGym/BigGymGameMode.cpp

```diff
@@ -17,8 +20,142 @@ ABigGymGameMode::ABigGymGameMode()
 	if (PlayerPawnBPClass.Class != NULL)
 	{
 		DefaultPawnClass = PlayerPawnBPClass.Class;
+		PlayerCheckoutRadius = FGenericPlatformMath::Sqrt(PlayerPawnBPClass.Class.GetDefaultObject()->NetCullDistanceSquared);
 	}
-	
+
+	PlayersSpawned = 0;
+
 	// Seamless Travel is not currently supported in SpatialOS [UNR-897]
 	bUseSeamlessTravel = false;
 }
+
+void ABigGymGameMode::InitGameState()
+{
+	if (ShouldUseCustomSpawning())
+	{
+		ParsePassedValues();
+		ClearExistingSpawnPoints();
+		GenerateSpawnPoints(TotalPlayers);
+	}
+}
+
+bool ABigGymGameMode::ShouldUseCustomSpawning()
+{
+	FString WorkerValue;
+	USpatialWorkerFlags::GetWorkerFlag(TEXT("override_spawning"), WorkerValue);
+	return (WorkerValue.Equals(TEXT("true"), ESearchCase::IgnoreCase) || FParse::Param(FCommandLine::Get(), TEXT("OverrideSpawning")));
+}
+
+void ABigGymGameMode::ParsePassedValues()
+{
+	if (FParse::Param(FCommandLine::Get(), TEXT("OverrideSpawning")))
+	{
+		if (FParse::Param(FCommandLine::Get(), TEXT("TotalPlayers=")))
+		{
+			FParse::Value(FCommandLine::Get(), TEXT("TotalPlayers="), TotalPlayers);
+		}
+	}
+	else
+	{
+		FString TotalPlayersString;
+
+		if (USpatialWorkerFlags::GetWorkerFlag(TEXT("total_players"), TotalPlayersString))
+		{
+			TotalPlayers = FCString::Atoi(*TotalPlayersString);
+		}
+	}
+}
+
+void ABigGymGameMode::ClearExistingSpawnPoints()
+{
+	TArray<AActor*> ExistingSpawnPoints;
+
+	UGameplayStatics::GetAllActorsOfClass(GetWorld(), APlayerStart::StaticClass(), ExistingSpawnPoints);
```

Comment:
Can't we use the SpawnPoints array here?

Reply:
Good suggestion, that should work. Changed it to use SpawnPoints, and added a `Reset` in `GenerateSpawnPoints` to clear out any spawnpoints collected here before adding new ones